src/lib/OS/SyscallHandler.cc

```diff
@@ -1,11 +1,10 @@
 #include "simeng/OS/SyscallHandler.hh"
 
+#include "simeng/OS/SimOS.hh"
 namespace simeng {
 namespace OS {
 
-SyscallHandler::SyscallHandler(
-    const std::unordered_map<uint64_t, std::shared_ptr<Process>>& processes)
-    : processes_(processes) {
+SyscallHandler::SyscallHandler(SimOS* os) : os_(os) {
```

Comment:
Okay sounds good to me. 

Just a quick note: I thought the whole point of having a Sim-OS object was to ensure that the core doesn't do any OS specific tasks? If the core needs the assistance of the OS, it can do so by invoking the OS (for SST through events via SST Links). Given the tight deadline we have and how messy circular dependencies can get, I understand the need to remove this in an upcoming PR so as to eliminate any extensive refactoring. However we need to ensure that the syscall handler can invoke the OS when needed. All files in the 'OS' folder collectively act as the OS for SimEng, SimOS is just as big a part of SyscallHandler as SyscallHandler is of SimOS. If the Syscall handler needs SimOS for something it should be able to invoke it. Existence of a circular dependency and multiple classes to achieve the functionality of a simulated OS are artifacts of OOP and the need to define a separation of responsibility between classes i.e SimOS and Syscall Handler.  However they both collectively come together to do act as the OS for SimEng. Same goes for Process.